### ISSUE TITLE
How to get property return type?

### ISSUE THREAD

**Issue by dmlittledm** [dotnet/runtime#59175]
I need to get return type of the property getter or a method.
I've got the TypeDefinition of required class, then got its properties and their getters' MethodDefinition-s
`var reader = module.GetMetadata().GetMetadataReader();`
`// ..`
`var typeDef = reader.GetTypeDefinition(tdHandle);`
`// ..`
`var properties = typeDef.GetProperties();`
`// ..`
`var property = reader.GetPropertyDefinition(propertyHandle);`
`var getter = property.GetAccessors().Getter;`
`var method = reader.GetMethodDefinition(getter);`

But how can I get the return types of these methods? 
I tried to get method implementations via `typeDef.GetMethodImplementations()` to map them with definitions but it returns empty collection.

Then I tried to decode method's Signature using code from Microsoft.Metadata.Tools,MetadataVisualizer but got something like "int16* (#172b)" as a result. And don't know what to do with it further. And this way to obtain method return type looks too complex and weird.

And I can't even google any suitable tips - most of answers are about just getting all the types from assembly.
Can you help me?

**Comment by teo-tsirpanis** [comment]
Hello, you have to call `DecodeSignature` with an `ISignatureTypeProvider` implementation of yours. What it would return depends on what kind if information you want to extract from a property's return type and on how you want them represented. `GetMethodImplementations` is totally unrelated to what you want to do.

`System.Reflection.Metadata` is indeed a pretty tough and low-level API, which is also what makes it decoupled and highly performant. If you want a heavier but much easier API to inspect assemblies using the traditional BCL reflection types, consider using [`System.Reflection.MetadataLoadContext`](https://docs.microsoft.com/en-us/dotnet/standard/assembly/inspect-contents-using-metadataloadcontext).

**Comment by buyaa-n** [comment]
Not sure if https://github.com/dotnet/roslyn/issues/14823#issuecomment-257355053 would help 
CC @tmat

**Comment by dmlittledm** [comment]
@teo-tsirpanis , @buyaa-n , thanks for your replies. Unfortunately they lead to the same place where I already were, when I've got method signature, that looks like "typedef#172" or "typedef#173<typeref#16b>".
But you inspired me for more experiments and I found that these hexadecimal numbers - 173 and 16b corresponds to index of types stored in MetadataReader's collections TypeDefinitions and TypeReferences. So I was able to obtain handles to these types this way:
`    var index = int.Parse(typeRefCode, NumberStyles.HexNumber);`
`    return reader.TypeReferences.Skip(index - 1).First();`

`    var index = int.Parse(typeDefCode, NumberStyles.HexNumber);`
`    return reader.TypeDefinitions.Skip(index - 1).First();`
